tests/testsuite.src/run_extensions.at

```diff
@@ -5601,3 +5600,143 @@ AT_CHECK([$COBCRUN_DIRECT ./prog], [0],
 ], [])
 
 AT_CLEANUP
+
+
+AT_SETUP([PICTURE L])
+AT_KEYWORDS([extensions gcos picture-l odo])
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 W-DATA.
+          2 X-DAT       PICTURE LX(10) DEPENDING ON LGX.
+          2 X-ALPHA     PICTURE LA(10) DEPENDING ON LGX.
+          2 Y-DAT       PICTURE LX(9) DEPENDING ON LGY.
+          2 Y-ALPHA     PICTURE LA(9) DEPENDING ON LGY.
+       1 W-LGS.
+          2 LGX         USAGE COMP-1 VALUE 10.
+          2 LGY         PIC 9 VALUE 9.
+       PROCEDURE        DIVISION.
+       MAIN.
+           MOVE SPACES TO W-DATA.
+           DISPLAY X-DAT "*" X-ALPHA "*" Y-DAT "*" Y-ALPHA "*"
+	   MOVE 5 TO LGX
+	   MOVE 1 TO LGY
+	   MOVE ".........." TO X-DAT
+	   MOVE "XXXXXXXXXX" TO X-ALPHA
+	   MOVE "YYYYYYYYY" TO Y-DAT
+	   MOVE "^^^^^^^^^" TO Y-ALPHA
+           DISPLAY X-DAT "*" X-ALPHA "*" Y-DAT "*" Y-ALPHA "*"
+	   MOVE X-DAT TO Y-DAT
+	   MOVE Y-ALPHA TO X-DAT
+           DISPLAY X-DAT "*" X-ALPHA "*" Y-DAT "*" Y-ALPHA "*"
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE prog.cob], [0], [], [])
+AT_CHECK([$COBCRUN_DIRECT ./prog], [0],
+[          *          *         *         *
+.....*XXXXX*Y*^*
+^    *XXXXX*.*^*
+])
+AT_CHECK([$COMPILE -fodoslide -o prog-x prog.cob], [0], [], [])
+AT_CHECK([$COBCRUN_DIRECT ./prog-x], [0],
+[          *          *         *         *
+.....*XXXXX*Y*^*
+^    *XXXXX*.*^*
+])
+
+
+AT_DATA([redefines.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      redefines.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 W-DATA.
+          2 UNUSED-1    PIC X VALUE "?".
+          2 X-DAT       PICTURE X(10).
+          2 Y1-DAT      REDEFINES X-DAT PICTURE LX(5) DEPENDING ON LGY.
+          2 Y2-DAT      REDEFINES X-DAT PICTURE LX(5) DEPENDING ON LGY.
+          2 UNUSED-2    PIC X VALUE "!".
+          2 Z-DAT       PICTURE LX(7) DEPENDING ON LGZ VALUE ".......".
+          2 T-DAT       REDEFINES Z-DAT PICTURE LX(7) DEPENDING ON LGT.
+          2 UNUSED-3    PIC X VALUE ".".
+       1 W-LGS.
+          2 LGX         PIC 99 VALUE 10.
+          2 LGY         PIC 9 VALUE 5.
+          2 LGZ         PIC 9 VALUE 3.
+          2 LGT         PIC 9 VALUE 5.
+       PROCEDURE        DIVISION.
+       MAIN.
+           MOVE "0123456789" TO X-DAT
+           DISPLAY X-DAT "*" Y1-DAT "*" Y2-DAT "*"
+           MOVE "0123456" TO Z-DAT
+           DISPLAY T-DAT "*" Z-DAT "*"
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE redefines.cob], [0], [], [])
+AT_CHECK([$COBCRUN_DIRECT ./redefines], [0],
+[0123456789*01234*01234*
+012..*012*
+])
+
+AT_CHECK([$COMPILE -fodoslide redefines.cob], [1], [],
+[redefines.cob:9: error: when ODOSLIDE is set, REDEFINES with PICTURE string with 'L' character are not yet supported
+])
+
+AT_DATA([prog_extraneous_depending.cob], [
```

Comment:
The test following are good, but should be moved to syn_misc.at, using `COMPILE_ONLY`.

There also should be a dialect option for `PIC L` (set to unconformable in all strict dialects, +warning in lax-inc, and ok for gcos and default).